### PR TITLE
G2 Electroshock Grenade "Sonic Electric Ball Breakers (SEBB)" [No Shitcode Edition]

### PR DESCRIPTION
**Pull request by PhantomEpicness**
# About the pull request
Adds the G2 electroshock grenade, "Sonic Electric ball breaker" as mentioned in the movies or SEBB for short. It's purpose is to be a tactical option for the marines, as it deals slowdown instead of stun/damage. (unless you stand on top of it)

Only meaningful damage is to marines and especially synths, since a 1.2 Gigavolt charge is probably not healthy for bioelectrics.

Can be used as a mine as well.

Made a PR so maintainers can take a look at it. Still WIP

# Explain why it's good for the game
More marine utility weapons and content, we can probably lower the amount of HEDPs slightly to balance it out aswell.

+ lore accurate af, as mentioned in the movie and the tech manual
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

https://github.com/cmss13-devs/cmss13/assets/42422230/a757dd2e-d29f-436a-a103-f4d0a7a2f4e2

image

# Changelog
:cl: TotalEpicness5, Triiodine, monkeyfist
add: Adds the G2 electroshock grenade. This grenade does not stun nor deal serious damage to xenos, but it stuns via electric shock. It can create EMP-like effects when faced with electronic devices like sentries, synths, Area Power Controllers, etc. It's damage is reduced by energy armor.
fix: take_overall_armoured_damage now can actually deal burn damage
balance: Marine armor has a small bit of energy armor added back. Heavy armor has slightly more protection than medium/light.
fix: Mines now check for explosive antigrief checks
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by PhantomEpicness** [comment]
rothschilds really paid off github to hide these conflicts before I submitted this

dmi conflicts will be resolved after code approval

**Comment by PhantomEpicness** [comment]
need advice on balancing

**Comment by FrogZeke** [comment]
Please correct the name in the PR:  Sonic Electronic Ball Breaker
https://avp.fandom.com/wiki/G2_Electroshock_Grenade

**Comment by Diegoflores31** [comment]
> need advice on balancing

Sprite & sound : the explosion effect looks a little goofy , the sound should be less noisy and preferably have less sounds to be in line with a normal grenade . just prime noise and them the boom.

balance : 
add FF ( a stun similar of touching a shocked door for marines in range) 
reduce the range of EMP effect.
make sure its Hand throwable only ( make it be the size of a M15 if possible so people dont confuse it with a M40).
reduce the 
for xenos , make it do a knockup effect on T1s and T2s only.
personally i dont like it taking the spot of Claymores . but i like it as a alternative grenade.

_all of this are suggestions i am not a maintaner_

**Comment by PhantomEpicness** [comment]
> > need advice on balancing
> 
> Sprite & sound : the explosion effect looks a little goofy , the sound should be less noisy and preferably have less sounds to be in line with a normal grenade . just prime noise and them the boom.
> 
> balance : add FF ( a stun similar of touching a shocked door for marines in range) reduce the range of EMP effect. make sure its Hand throwable only ( make it be the size of a M15 if possible so people dont confuse it with a M40). reduce the for xenos , make it do a knockup effect on T1s and T2s only. personally i dont like it taking the spot of Claymores . but i like it as a alternative grenade.
> 
> _all of this are suggestions i am not a maintaner_

I forgot to mention this in the description but the damage it deals is very substantial for marines (but miniscule for xenos less they are ontop of it for some reason). Most damage also goes straight through armor as it deals energy damage. I'll post a video later when I make the thread on discord

In addition synths get stunned + 2x damage (1.2 GV ain't good for electrical anything) and "resets" the fire control systems of sentries by setting its range to 0 for a few seconds.

I will add some more negative effects for marines though, perhaps a flicker stun "every muscle in your body seizes up momentarily" and might adjust the sentry disable time too

**Comment by PhantomEpicness** [comment]
> Interesting bug with req vendors
> 
>  ow.mp4

real

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by PhantomEpicness** [comment]
fucked up the rebase hold on

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by PhantomEpicness** [comment]
ok cleared

**Review by sleepynecrons — APPROVED** [review]
sprites good

**Comment by PhantomEpicness** [comment]
changes:
uses stamina damage for humans
mild damage buff since you would get like 10 damage from 3 tiles away before
code readability improvements and variable updates.
removed stupid sparks
 
damage equation:

damage applied = damage - (distance to nade falloff factor)* 
falloff factor is 12
damage is 120
so if a marine is 3 tiles away
3 * 12 = 36
120 damage - 36 falloff = 84
which then faces ARMOR_ENERGY reduction and spread across all limbs 

todo: damage testing

**Review by harryob — CHANGES_REQUESTED** [review]
failing tests, undraft once you've fixed them

**Review by SabreML — CHANGES_REQUESTED** [review]
A few things, looking good though!

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/game/objects/items/explosives/grenades/marines.dm
Comment (PhantomEpicness) [reply code/game/objects/items/explosives/grenades/marines.dm:498]: will do

**Comment by PhantomEpicness** [comment]
icon conflicts will be resolved when code/balance approval is done

File: code/game/objects/items/explosives/mine.dm
Comment (PhantomEpicness) [reply code/game/objects/items/explosives/mine.dm:326]: was testing something

nice catch

File: code/game/objects/items/explosives/grenades/marines.dm
Comment (PhantomEpicness) [reply code/game/objects/items/explosives/grenades/marines.dm:507]: fixed

**Comment by PhantomEpicness** [comment]
changes:
-Falloff factor is now 20
-damage reduced to 110
example: a human 3 tiles away will only recieve 50 damage *before armor calculations*
3*20 = 60
110 - 60 = 50
- general refractor so balance numbers are easier to modify
- tuned down human stamina damage a bit
- lowered mine detonation time to 0.7 seconds so it still can catch runners
- now deals actual burn damage

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by PhantomEpicness** [comment]
Marine energy armor return:
Medium and light armor: LOW
Heavy: MEDIUMLOW
B18, nade spec armor: MEDIUM

File: code/game/objects/items/explosives/grenades/marines.dm
Comment (Drulikar) [inline code/game/objects/items/explosives/grenades/marines.dm:508]: If a synth can't prime it, I don't think they should be able to plant it.

Replicate `Your programming prevents you from using this!` here.

File: code/game/objects/items/explosives/grenades/marines.dm
Comment (Drulikar) [inline code/game/objects/items/explosives/grenades/marines.dm:525]: See `/obj/item/explosive/mine/Initialize()` where it for whatever reason is calling `deploy_mine(null)` because it has map_deployed true. This results in these basically having their IFF hardcoded to USCM. You likely want to do it more like how `/obj/item/explosive/mine/attack_self(mob/living/user)` does it.

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by deathrobotpunch** [comment]
marine balls shall be broken

**Comment by PhantomEpicness** [comment]
balance questions to be solved with a tm:

- availability
- slowdown amounts

File: code/game/objects/items/explosives/grenades/marines.dm
Comment (PhantomEpicness) [reply code/game/objects/items/explosives/grenades/marines.dm:531]: IFF is set when planting now.

File: code/game/objects/items/explosives/grenades/marines.dm
Comment (PhantomEpicness) [reply code/game/objects/items/explosives/grenades/marines.dm:514]: fixed

**Review by Drulikar — CHANGES_REQUESTED** [review]
Couple more issues before I can TM this:
- It can be planted w/o adhering to safe-area accident inhibitor (uh claymores too...) this isn't *too* much of an issue if there isn't a way to detonate them but it would likely be weird w/ survivors that go shipside. Technically isn't as relevant currently on live, but I would prefer all mines not be plantable wherever/whenever grenades can't be primed.
- When you plant it, its not anchored, can be pulled around, and be picked up. If you pick it up, then you can deploy it like a claymore. It should instead behave like a deployed claymore when planted.

**Comment by Superwelder** [comment]
image


Examine info text does not contain the timer amount unlike HEDP or M15s.

File: code/game/objects/items/explosives/grenades/marines.dm
Comment (Drulikar) [inline code/game/objects/items/explosives/grenades/marines.dm:551]:
```suggestion
		playsound(loc, 'sound/effects/sebb_explode.ogg', 90, 0, 10)
	else
		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), loc, 'sound/effects/sebb_beep.ogg', 60, 0, 10), soundtime)
```

**Review by Drulikar — CHANGES_REQUESTED** [review]
Be sure to install the tools/hooks to resolve the DMI conflict as well (before merging upstream master into your branch locally).

**Pull request closed without merging**
